benchmark.py

```diff
@@ -3,10 +3,346 @@
+    )
+    parser.add_argument(
+        '--sg', default=1, choices=[0, 1], type=int,
+        help='Use the skip-gram model; default is %(default)s (use 0 for continuous bag of words model)'
+    )
+    parser.add_argument(
+        '--negative', default=5, type=int,
+        help='Number of negative examples; default is %(default)s, common values are 3 - 10 (0 = not used)'
+    )
+    parser.add_argument(
+        '--batch_size', default=32, type=int,
+        help='Mini batch size for training. Default : %(default)s'
+    )
+    parser.add_argument(
+        '--alpha', default=0.025, type=float,
+        help='The initial learning rate. Default : %(default)s'
+    )
+    parser.add_argument(
+        '--platform', help='Platform the benchmark is being run on. eg. aws, azure', required=True
+    )
+    return parser.parse_args()
+
+
+def prepare_params(options):
+    params = vars(options).copy()  # ensure original options not modified
+    params.pop('frameworks')
+    params['outputpath'] = '{}/{}'.format(os.getcwd(), TRAINED_VEC_SAVE_DIR)
+    params.pop('platform')
+    return params
+
+
+def check_gpu():
+    try:
+        check_output('nvidia-smi', shell=True)
+        return 1
+    except:
+        return 0
 
-from hyperparams import Hyperparams as hp
 
 if __name__ == '__main__':
-    train = Train(epochs=10)
-    train.train_gensim()
+
+    options = parse_args()
+    report_dict = dict()
+
+    TRAINED_VEC_SAVE_DIR = 'trainedVectors/'
+    QA_FILE_PATH = 'data/questions-words.txt'
+    WORD_PAIRS_DIR = 'data/word-sim/'
+    REPORT_FILE = "{}-report.json".format(options.platform)
+
+    GPU = check_gpu()  # indicates if gpu capability exists
+    FRAMEWORKS_GPU = ['tensorflow']
+
+    # get params required for training
+    params = prepare_params(options)
+    train = Train(**params)
+    clear_trained_vecs(REPORT_FILE, TRAINED_VEC_SAVE_DIR)
+
+    # store system information
+    report_dict['systeminfo'] = get_cpu_info()
+
+    # store gpu information, if gpu capability exists
+    if GPU:
+        report_dict['systeminfo'] += '\n{}'.format(get_gpu_info())
+
+    # write config_str/model parameters to a file - useful for showing training params in the final plots
+    report_dict['trainingparams'] = vars(options)
+    if GPU and 'tensorflow' in options.frameworks:
+        report_dict['frameworks'] = options.frameworks + ['tensorflow-gpu']
+    else:
+        report_dict['frameworks'] = options.frameworks
+    report_dict['platform'] = options.platform
+    report_dict['time'] = dict()
+    report_dict['memory'] = dict()
+    report_dict['command'] = dict()
+    report_dict['wordpairs'] = dict()
+    report_dict['qa'] = dict()
+
+    # train and evaluate one framework at a time
+    for framework in options.frameworks:
+        report_dict['wordpairs'][framework] = []
+        report_dict['qa'][framework] = []
+        report_dict = train.train_framework(framework, 0, report_dict)
```

Comment:
Please don't pass `report_dict` to functions, you should return new dict from function and merge it with current `report_dict` (in `train_frameword`, `eval_word_vectors`)